evaluation/scripts/wmt.py

```diff
@@ -0,0 +1,52 @@
+import argparse
+
+import torch
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from ..datasets.wmt import WMTEnglishDataset
+
+
+def main(args):
+    """adapted from https://huggingface.co/transformers/perplexity.html"""
+    stride = args.stride
+    device = torch.device(args.device)
+    tokenizer = AutoTokenizer.from_pretrained(args.model)
+    model = AutoModelForCausalLM.from_pretrained(args.model).to(device)
+    dataset = WMTEnglishDataset(
+        tokenizer,
+        stride=stride,
+        max_len=model.config.n_positions,
+        pair=args.pair,
+    )
+    loader = DataLoader(
+        dataset,
+        batch_size=args.batch_size,
+        num_workers=args.num_workers,
+        shuffle=False,
+        drop_last=True,
+    )
+    log_likelihoods = []
+    for input_ids in tqdm(loader):
+        input_ids = input_ids.to(device)
+        target_ids = input_ids.clone()
+        target_ids[:, :-stride] = -100
+        with torch.no_grad():
+            outputs = model(input_ids, labels=target_ids)
+            log_likelihood = outputs[0]
+        log_likelihoods.append(log_likelihood)
+    perplexity = torch.exp(torch.stack(log_likelihoods).sum() / len(loader))
```

Comment:
If the above suggestion is taken, then this line needs a tweak. `num_targets` is the total number of target words.

```suggestion
    perplexity = torch.exp(torch.stack(log_likelihoods).sum() / num_targets)
```